hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/metrics/TestS3GatewayMetrics.java

```diff
@@ -110,4 +123,160 @@ public void testHeadObject() throws Exception {
+
+    long oriMetric = metrics.getCreateBucketSuccess();
+
+    bucketEndpoint.put(bucketName, null,
+        null, null);
+    long curMetric = metrics.getCreateBucketSuccess();
+
+    assertEquals(1L, curMetric - oriMetric);
+  }
+
+  @Test
+  public void testCreateBucketFailure() throws Exception {
+    // Creating an error by trying to create a bucket that already exists
+    long oriMetric = metrics.getCreateBucketFailure();
+
+    bucketEndpoint.put(bucketName, null, null, null);
+
+    long curMetric = metrics.getCreateBucketFailure();
+    assertEquals(1L, curMetric - oriMetric);
+  }
+
+  @Test
+  public void testDeleteBucketSuccess() throws Exception {
+    long oriMetric = metrics.getDeleteBucketSuccess();
+
+    bucketEndpoint.delete(bucketName);
+
+    long curMetric = metrics.getDeleteBucketSuccess();
+    assertEquals(1L, curMetric - oriMetric);
+  }
+
+  @Test
+  public void testDeleteBucketFailure() throws Exception {
+    long oriMetric = metrics.getDeleteBucketFailure();
+    bucketEndpoint.delete(bucketName);
+    try {
+      // Deleting a bucket that does not exist will result in delete failure
+      bucketEndpoint.delete(bucketName);
+      fail();
+    } catch (OS3Exception ex) {
+      assertEquals(S3ErrorTable.NO_SUCH_BUCKET.getCode(), ex.getCode());
+      assertEquals(S3ErrorTable.NO_SUCH_BUCKET.getErrorMessage(),
+          ex.getErrorMessage());
+    }
+
+    long curMetric = metrics.getDeleteBucketFailure();
+    assertEquals(1L, curMetric - oriMetric);
+  }
+
+  @Test
+  public void testGetAclSuccess() throws Exception {
+    long oriMetric = metrics.getGetAclSuccess();
+
+    Response response =
+        bucketEndpoint.get(bucketName, null, null,
+            null, 0, null, null,
+            null, null, "acl", null);
+    long curMetric = metrics.getGetAclSuccess();
+    assertEquals(HTTP_OK, response.getStatus());
+    assertEquals(1L, curMetric - oriMetric);
+  }
+
+  @Test
+  public void testGetAclFailure() throws Exception {
+    long oriMetric = metrics.getGetAclFailure();
+    try {
+      // Failing the getACL endpoint by applying ACL on a non-Existent Bucket
+      bucketEndpoint.get("random_bucket", null,
+          null, null, 0, null,
+          null, null, null, "acl", null);
+      fail();
+    } catch (OS3Exception ex) {
+      assertEquals(S3ErrorTable.NO_SUCH_BUCKET.getCode(), ex.getCode());
+      assertEquals(S3ErrorTable.NO_SUCH_BUCKET.getErrorMessage(),
+          ex.getErrorMessage());
+    }
+    long curMetric = metrics.getGetAclFailure();
+    assertEquals(1L, curMetric - oriMetric);
+  }
+
+  @Test
+  public void testPutAclSuccess() throws Exception {
+    long oriMetric = metrics.getPutAclSuccess();
+
+    clientStub.getObjectStore().createS3Bucket("b1");
+    InputStream inputBody = TestBucketAcl.class.getClassLoader()
+        .getResourceAsStream("userAccessControlList.xml");
+
+    bucketEndpoint.put("b1", ACL_MARKER, headers, inputBody);
+    inputBody.close();
```

Comment:
Nit: using try-with-resources we can ensure that the stream is closed even in case of failure.

```suggestion
    try (InputStream inputBody = TestBucketAcl.class.getClassLoader()
          .getResourceAsStream("userAccessControlList.xml")) {

      bucketEndpoint.put("b1", ACL_MARKER, headers, inputBody);
    }
```